packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -447,6 +444,36 @@ export class Pipeline extends PipelineBase {
     return !owner || owner === 'AWS';
   }
 
+  private getOtherStackIfActionIsCrossAccount(action: IAction): Stack | undefined {
+    const pipelineStack = Stack.of(this);
+
+    if (action.actionProperties.resource) {
```

Comment:
What is the expected behavior if account is AWS::AccountId token? Maybe we should block?

Reply:
I'm not sure actually. I assume the following:

* If both Stack use `AWS::AccountId` (i.e., no explicit `env` was provided), then they will be equal, so nothing will happen.
* If one of the accounts has been provided, and the other not, the check for `!==` will succeed, and will trigger the cross-account behavior.

What do you mean by "block"?

Reply:
I meant that maybe we should demand that if you are specifying an account ID (which must be concrete) as the target for the action, the source stack will also have to use an explicit account ID (it can't be env-agnostic).

Reply:
Done.